### DISCUSSION TITLE
Intellisense dramatically slows after including Shiny

### DISCUSSION THREAD

**Discussion by mphill** [shinyorg/shiny#956] [Q&A]
**Platforms Affected:** Visual Studio Mac 8.10.20 and prior

It might be Windows too, however I don't have any Windows machines to test with.  And if it was Windows more people would be complaining I am sure.

I've tried to use Shiny.BLE in 2 separate projects over the past 8 months.  After including Shiny, the IDE's intellisense starts to take takes 1-5+ seconds when previously it would be virtually instantaneous.  After removing Shiny the responsiveness comes back.

Perhaps is an issue with the source code generators and VS for Mac. I can reliably reproduce this issue though.

Thanks

**Comment by aritchie** [comment]
I have not noticed this myself, but VS4mac is always problematic no matter what I am doing.  There are certainly points where the source generators can be perf improved, but this isn't something I'm going to do since it is only used for boilerplate code insertion.  It isn't needed to increase overall app performance and the user can insert the boilerplate code themselves which is what I would recommend doing if the perf is bothering you.

**Reply by mphill** [reply]
@aritchie Do you actively develop this library on a Mac or Windows?  I agree, VS4Mac can be soul crushing...  

It does seem to work much faster on VS 2022, but hot reload isn't functional there yet.  Perhaps this issue resolves once 2022 is released officially for the Mac.
